src/editors/_baseEditor.js

```diff
@@ -12,258 +14,340 @@ export const EditorState = {
+      selection = this.instance.getSelectedLast();
+
+      if (selection[0] > selection[2]) {
+        tmp = selection[0];
+        selection[0] = selection[2];
+        selection[2] = tmp;
+      }
+      if (selection[1] > selection[3]) {
+        tmp = selection[1];
+        selection[1] = selection[3];
+        selection[3] = tmp;
+      }
+    } else {
+      selection = [this.row, this.col, null, null];
+    }
 
-    return Child;
+    this.instance.populateFromArray(selection[0], selection[1], value, selection[2], selection[3], 'edit');
   }
 
-  return inherit(Editor, baseClass);
-};
-
-BaseEditor.prototype.saveValue = function(value, ctrlDown) {
-  let selection;
-  let tmp;
+  /**
+   * Begins editing on a highlighted cell and hides fillHandle corner if was present.
+   *
+   * @param {*} newInitialValue
+   * @param {*} event
+   */
+  beginEditing(newInitialValue, event) {
+    if (this.state !== EditorState.VIRGIN) {
+      return;
+    }
+    this.instance.view.scrollViewport(new CellCoords(this.row, this.col));
+    this.state = EditorState.EDITING;
 
-  // if ctrl+enter and multiple cells selected, behave like Excel (finish editing and apply to all cells)
-  if (ctrlDown) {
-    selection = this.instance.getSelectedLast();
+    // Set the editor value only in the full edit mode. In other mode the focusable element has to be empty,
+    // otherwise IME (editor for Asia users) doesn't work.
+    if (this.isInFullEditMode()) {
+      const stringifiedInitialValue = typeof newInitialValue === 'string' ? newInitialValue : stringify(this.originalValue);
 
-    if (selection[0] > selection[2]) {
-      tmp = selection[0];
-      selection[0] = selection[2];
-      selection[2] = tmp;
-    }
-    if (selection[1] > selection[3]) {
-      tmp = selection[1];
-      selection[1] = selection[3];
-      selection[3] = tmp;
+      this.setValue(stringifiedInitialValue);
     }
-  } else {
-    selection = [this.row, this.col, null, null];
-  }
 
-  this.instance.populateFromArray(selection[0], selection[1], value, selection[2], selection[3], 'edit');
-};
-
-BaseEditor.prototype.beginEditing = function(newInitialValue, event) {
-  if (this.state !== EditorState.VIRGIN) {
-    return;
-  }
-  this.instance.view.scrollViewport(new CellCoords(this.row, this.col));
-  this.state = EditorState.EDITING;
+    this.open(event);
+    this._opened = true;
+    this.focus();
 
-  // Set the editor value only in the full edit mode. In other mode the focusable element has to be empty,
-  // otherwise IME (editor for Asia users) doesn't work.
-  if (this.isInFullEditMode()) {
-    const stringifiedInitialValue = typeof newInitialValue === 'string' ? newInitialValue : stringify(this.originalValue);
+    // only rerender the selections (FillHandle should disappear when beginediting is triggered)
+    this.instance.view.render();
 
-    this.setValue(stringifiedInitialValue);
+    this.instance.runHooks('afterBeginEditing', this.row, this.col);
```

Comment:
As the `this.instance` property is marked as `@deprecated` could you rewrite to a new property (`this.hot`)?